src/bin/worker.js

```diff
@@ -122,6 +119,16 @@ program.parse(process.argv);
     config = _.defaultsDeep(targetConfig, config);
   }
 
+  process.on('unhandledRejection', async (reason, p) => {
+    console.error(`Unhandled rejection at ${p}.\n${reason.stack || reason}`);
+    if (host && host.shutdown) {
+      await host.shutdown();
+    } else {
+      spawn('shutdown', ['-h', 'now']);
```

Comment:
This might be helpful if an interactive task is running (so that the interactive session is alerted to the reason for the shutdown):

```suggestion
      spawn('shutdown', ['-h', 'now', `docker-worker shutdown due to unhandled rejection ${p}`]);
```